### PR TITLE
sys-apps/iproute2: Fix building on musl profile

### PR DESCRIPTION
**Pull request by listout**
Closes: https://bugs.gentoo.org/926341
Closes: https://bugs.gentoo.org/936234

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by floppym** [comment]
Please send patches upstream first. We can backport from there.

**Comment by listout** [comment]
I did some time ago, but probably it got lost or something https://lore.kernel.org/netdev/20240514063811.383371-1-brahmajit.xyz@gmail.com/T/#u

File: sys-apps/iproute2/files/iproute2-6.10.0-fix-implecit-declaration-of-basename-on-musl.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Bug: https://bugs.gentoo.org/926341
+From: Brahmajit Das <brahmajit.xyz@gmail.com>
+Date: Sun, 21 Jul 2024 07:31:49 +0000
+Subject: [PATCH 2/2] Fix implecit declaration of basename on musl
+
+On musl there is only one implementation of musl which comes from
```
Comment (floppym) [inline sys-apps/iproute2/files/iproute2-6.10.0-fix-implecit-declaration-of-basename-on-musl.patch:6]:
```suggestion
On musl there is only one implementation of basename which comes from
```

File: sys-apps/iproute2/files/iproute2-6.10.0-fix-implecit-declaration-of-basename-on-musl.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+libgen.h, unlike glibc which has two implementations. I'm not really
+sure if using the basename from libgen.h is correct over here i.e. using
+the posix implementation of basename.
```
Comment (floppym) [inline sys-apps/iproute2/files/iproute2-6.10.0-fix-implecit-declaration-of-basename-on-musl.patch:9]: If you are not sure the POSIX implementation is correct here, please look into it and confirm it.

**Review by floppym — CHANGES_REQUESTED** [review]
Please send a follow-up message to the upstream maintainers.

**Comment by hhoffstaette** [comment]
There are exactly two calls to `basename`:
```
$grep -r 'basename('
lib/bpf_legacy.c:			 basename(strdupa(cfg->object)),
rdma/rdma.c:	filename = basename(argv[0]);
```
`strdupa()` makes potential mutation safe, and AFAIK elements of argv can be mutated due to being declared non-const historically. Apparently [C99 also says this](https://stackoverflow.com/a/963777/99851). I don't see why applying the patch using libgen.h should not work.

**Comment by floppym** [comment]
> `strdupa()` makes potential mutation safe, and AFAIK elements of `argv` can be mutated due to historically being declared non-const. Apparently [C99 also says this](https://stackoverflow.com/a/963777/99851). I don't see why applying the patch using libgen.h should not work.

Thanks, that's good enough for me.

**Pull request closed without merging**